src/platform/input/game-pads.js

```diff
@@ -134,111 +935,142 @@ class GamePads {
-        return !this.current[index].pad.buttons[i].pressed && (this.previous[index] && this.previous[index][i]);
+    /**
+     * Make the gamepad vibrate.
+     *
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
+     * @param {number} intensity - Intensity for the vibration in the range 0 to 1.
+     * @param {number} duration - Duration for the vibration in milliseconds.
+     * @param {object} [options] - Options for special vibration pattern.
+     * @param {number} [options.startDelay] - Delay before the pattern starts, in milliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators in the range 0 to 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators in the range 0 to 1. Defaults to intensity.
+     * @returns {Promise<boolean>} Return a Promise resulting in true if the pulse was successfully completed.
+     */
+    pulse(orderIndex, intensity, duration, options) {
+        const pad = this.current[orderIndex];
+        return pad ? pad.pulse(intensity, duration, options) : Promise.resolve(false);
     }
 
     /**
-     * Get the value of one of the analogue axes of the pad.
+     * Make all gamepads vibrate.
      *
-     * @param {number} index - The index of the pad to check, use constants {@link PAD_1},
-     * {@link PAD_2}, etc.
-     * @param {number} axes - The axes to get the value of, use constants {@link PAD_L_STICK_X},
-     * etc.
-     * @returns {number} The value of the axis between -1 and 1.
+     * @param {number} intensity - Intensity for the vibration in the range 0 to 1.
+     * @param {number} duration - Duration for the vibration in milliseconds.
+     * @param {object} [options] - Options for special vibration pattern.
+     * @param {number} [options.startDelay] - Delay before the pattern starts, in milliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators in the range 0 to 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators in the range 0 to 1. Defaults to intensity.
+     * @returns {Promise<boolean[]>} Return a Promise resulting in an array of booleans defining if the pulse was successfully completed for every gamepads.
      */
-    getAxis(index, axes) {
-        if (!this.current[index]) {
-            return 0;
-        }
+    pulseAll(intensity, duration, options) {
+        return Promise.all(
+            this.current.map(pad => pad.pulse(intensity, duration, options))
+        );
+    }
 
-        const key = this.current[index].map.axes[axes];
-        let value = this.current[index].pad.axes[pc[key]];
+    /**
+     * Find a connected {@link GamePad} from its identifier.
+     *
+     * @param {string} id - The identifier to search for.
```

Comment:
Worth enumerating the valid values here? It's not obvious what the ids are.

Reply:
What do you mean? Id is defined by constructor, it's the name of the gamepad so there is no real value to enumerate...

Reply:
Oh right. I thought this took `PAD_0`, PAD_1` etc. Fair enough.